fed/_private/compatible_utils.py

```diff
@@ -152,12 +153,28 @@ def reset(self):
 
 def _init_internal_kv():
     """An internal API that initialize the internal kv object."""
-    from ray._private.client_mode_hook import is_client_mode_enabled
-    if is_client_mode_enabled:
-        kv_actor = ray.remote(InternalKv).options(name="_INTERNAL_KV_ACTOR").remote()
-        response = kv_actor._ping.remote()
-        ray.get(response)
-    return ClientModeInternalKv() if is_client_mode_enabled else InternalKv()
-
-
-kv = _init_internal_kv()
+    global kv
+    if kv is None:
+        from ray._private.client_mode_hook import is_client_mode_enabled
+        if is_client_mode_enabled:
+            kv_actor = ray.remote(InternalKv) \
+                .options(name="_INTERNAL_KV_ACTOR").remote()
+            response = kv_actor._ping.remote()
+            ray.get(response)
+        kv = ClientModeInternalKv() if is_client_mode_enabled else InternalKv()
+
+
+def _clear_internal_kv():
+    global kv
+    if kv is not None:
+        kv.delete(constants.KEY_OF_CLUSTER_CONFIG)
+        kv.delete(constants.KEY_OF_JOB_CONFIG)
+        kv.reset()
+        from ray._private.client_mode_hook import is_client_mode_enabled
+        if is_client_mode_enabled:
+            _internal_kv_actor = ray.get_actor("_INTERNAL_KV_ACTOR")
+            ray.kill(_internal_kv_actor)
```

Comment:
kill is an asynchronized operation, what if it gets failed or should we wait until it gets killed?

Reply:
Agreed, also, it would be really nice that we wait unitil recv/send proxy actor get killed

```
class ProxyActor:
  def shutdown(self):
    ray.actor.exit_actor()

try:
  ray.get(proxy_handle.shutdown.remote())
except RayActorError:
  logger.info('Actor cleaned')
else:
  logger.error('Actor force clean error')
```

Reply:
True.  But another thing is that

accessing internal_kv in client mode has an official interface (under *python/ray/util/client*), there's actually no need to create an actor and manage its lifecycle.  So the current implementation may get deprecated soon.